capstone/src/main/webapp/authenticate.js

```diff
@@ -0,0 +1,89 @@
+var CLIENT_ID = config.CLIENT_ID;
+var API_KEY = config.API_KEY;
+var DISCOVERY_DOCS = ["https://script.googleapis.com/$discovery/rest?version=v1"];
+var SCOPES = 'https://www.googleapis.com/auth/forms';
+
+document.getElementById("auth").appendChild(createLoginButton());
+document.getElementById("auth").appendChild(createLogoutButton());
+var authorizeButton = document.getElementById('authorize_button');
+var signoutButton = document.getElementById('signout_button');
```

Comment:
IDs are not a preferred way of getting things by since they could collide with other things on the page. Also, this relies on there being an existing element called "auth" which may or may not already have content. Creating the buttons yourself is good, but you should also create the entire auth button element thingy. You can have that function return the relevant elements and not need to use any IDs at all.
// At this location
var [dynamicButton, authorizeButton, signoutButton] = createDynamicSigninButton();
// Other JS which brings this script in will decide itself how it wants to use the dynamicButton variable. It will probably have a call in its own stuff which is something to the effect of:
document.getElementById("auth_placeholder").replace_with(dynamicButton);


function createDynamicSigninButton() {
     var dynamicButton = document.createElement("div");
     var loginButton = createLoginButton();
    dynamicButton.appendChild(loginButton);
    // same for logout
    return [dynamicButton, loginButton, logoutButton];
}

Reply:
Also reasonable to inline the createXButton calls since they are only relevant in this one context

Reply:
Also, it would be preferable to encapsulate all of this into one "component" (not exactly in the framework sense, but conceptually), but that is a bit more in-depth discussion. Remind me in stand-up

Reply:
Got it. Resolved.